### PR TITLE
parser: support plan change capture collect stmt

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/41130

Problem Summary:

### What is changed and how it works?

support plan change capture collect stmt

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -14730,6 +14731,23 @@ PlanReplayerStmt:
...
 		}
 
+		$$ = x
+	}
+
+/********************************************************************
+ *
+ * Plan Chnage Capture Statement
+ *
+ * PLAN REPLAYER Change 'begin_time' 'end_time'
```
Comment (qw4990) [inline parser/parser.y:14741]: Can we use `now()` as the `end_time` here?

Comment (qw4990) [reply parser/parser.y:14741]: And `plan change capture now()-3d, now()` is ok?

Comment (Yisaer) [reply parser/parser.y:14741]: I think we can make `plan change capture colloect <ts1>` as `plan change capture collect <ts1> now()`, but I think support `now()-3d` is not necessary.

File: executor/builder.go [hunk 2]
```diff
@@ -1017,6 +1017,25 @@ func (b *executorBuilder) buildIndexAdvise(v *plannercore.IndexAdvise) Executor
 	return e
 }
 
+func (b *executorBuilder) buildPlanChangeCapture(v *plannercore.PlanChangeCapture) Executor {
+	e := &PlanChangeCaptureExec{
+		baseExecutor: newBaseExecutor(b.ctx, v.Schema(), v.ID()),
+	}
+	begin, err := time.Parse("2006-01-02 15:04:05", v.Begin)
```
Comment (qw4990) [inline executor/builder.go:1024]: Should we consider time zone here?

**Review by qw4990 — APPROVED** [review]

File: executor/builder.go [hunk 3]
```diff
@@ -1017,6 +1017,25 @@ func (b *executorBuilder) buildIndexAdvise(v *plannercore.IndexAdvise) Executor
...
+	if err != nil {
+		e.err = err
+		return e
+	}
+	end, err := time.Parse("2006-01-02 15:04:05", v.End)
```
Comment (xuyifangreeneyes) [inline executor/builder.go:1029]: Maybe we can use `types.TimeFormat` instead?

**Pull request merged**